syntax = "proto3";
package org.example.grpc.Calculator;

enum Operator {
  ADD = 0;
  SUB = 1;
  MUL = 2;
  DIV = 3;
}

message BinaryOpRequest {
  Operator operator = 1;
  float operandA = 2;
  float operandB = 3;
}

message BinaryOpResponse {
  float result = 1;
}

service Calculator {
  rpc calculate(BinaryOpRequest) returns (BinaryOpResponse) {}
}


